include/openssl/ssl.h

```diff
@@ -1626,6 +1627,8 @@ __owur int SSL_add_file_cert_subjects_to_stack(STACK_OF(X509_NAME) *stackCAs,
                                                const char *file);
 int SSL_add_dir_cert_subjects_to_stack(STACK_OF(X509_NAME) *stackCAs,
                                        const char *dir);
+int SSL_add_store_cert_subjects_to_stack(STACK_OF(X509_NAME) *stackCAs,
```

Comment:
I still think this should not be public

Reply:
Considering the old ones are public (and should be deprecated), I don't understand why not.

Reply:
Why does it need to be public? The old ones have the feeling of an internal function that were accidentally made public. Every time we add a public function it brings with it a support burden. We should only add public functions that are actually needed. I am not at all convinced that this is the case here. Just because we have other similar functions, that is not a good enough reason to add another one.

Reply:
I don't at all think they are accidentally public, they're part of a group of functions that users can use to build a stack of CA names for specifying acceptable peer cert issuers.  As for our own use, we do use `SSL_add_file_cert_subjects_to_stack()` in s_client.c (why we don't have a `-ReqCApath` option, *I do not know*...  *That*, I'm ready to believe is an accidental omission):

https://github.com/openssl/openssl/blob/6624e1f7b6a397948561e9cc2774f0c8af1d2c79/apps/s_client.c#L1808

But think of it another way...  Debian has a `ca-certificates` package, which populates `/etc/ssl/certs` with a number of certificates to be used by anyone that trusts Debian to do the right thing (they are quite diligent), and that directory is the perfect candidate for a `SSL_add_dir_cert_subjects_to_stack()` call.

I do not know, and have no intention to search for how many application actually use these particular functions.  But I believe they are intentionally public, and with that in mind, it would be *wrong* not to make the store one public as well.

Reply:
Ok. That's a fair argument.